### PR TITLE
[Storage] az storage queue list: Track2 supported

### PR DESCRIPTION
**Pull request by evelyn-ys**
## Description ##
Migrate `az storage queue` to track2 sdk. Only `list` command is implemented in main repo, others will be developed in azure-cli-extension.

## Testing Guide ##
`az storage queue list --account-name $STORAGE_ACCOUNT`

### CODE REVIEW

**Comment by yonzhan** [comment]
Storage

**Review by Juliehzl — CHANGES_REQUESTED** [review]
could you also add test for account key and connection string scenarios?

File: src/azure-cli/azure/cli/command_modules/storage/_help.py [hunk 1]
```diff
@@ -2247,12 +2250,3 @@
 type: group
 short-summary: Manage shared access policies of a storage table.
 """
-
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/storage/_help.py:2250]: why removing command group description here?

Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/storage/_help.py:2250]: > why removing command group description here?

This was copied from 'storage table policy' whose summary not suitable for 'storage queue'.
Actually we already have 'storage queue' description from line 2023 to line 2026

https://github.com/Azure/azure-cli/pull/15494/files/144c774e1877079a962235efe30cb3bcc233b971#diff-d6b7a29513fb666c1dee4ee655c8af9ab67ab76066278ad4ebc73a457a970d88#L2023-L2026

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 2]
```diff
@@ -1341,6 +1341,16 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
...
+    with self.argument_context('storage queue list') as c:
+        c.argument('include_metadata', help='Specifies that queue metadata be returned in the response.')
+        c.argument('marker', arg_type=marker_type)
+        c.argument('num_results', arg_type=num_results_type)
+        c.argument('prefix', help='Filters the results to return only queues whose names '
+                                  'begin with the specified prefix.')
+        c.argument('show_next_marker', action='store_true',
+                   help='Show nextMarker in result when specified.')
+        c.extra('timeout', help='Request timeout in seconds. Applies to each call to the service.', type=int)
+
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py:1353]: Use active voice

**Review by Juliehzl — APPROVED** [review]

**Pull request merged**